addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1099,21 +1105,23 @@ var SnippetEditor = Widget.extend({
         // Remove the siblings that belong to a snippet in grid mode
         // and put the identified grid mode snippets in their own "selector".
         const selectorGrids = new Set();
-        if (rowEl.classList.contains('row')) {
-            if ($selectorSiblings) {
-                // Looping backwards because elements are removed, so the
-                // indexes are not lost.
-                for (let i = $selectorSiblings.length - 1; i >= 0; i--) {
-                    if (isPopup && !$selectorSiblings[i].closest('div.s_popup')) {
-                        // Removing the siblings that are outside the popup if
-                        // the grid item is in a popup.
+        if (rowEl.classList.contains('row') && $selectorSiblings) {
+            // The grid from which the dragged column comes (if any).
+            const targetGridEl = rowEl.classList.contains('o_grid_mode') ? rowEl : rowEl.closest('.o_grid_mode');
```

Comment:
1. If I have an outer snippet which is in grid mode and `rowEl` here is from an inner snippet which is not in grid mode, is not that comment wrong? Maybe the code is necessary but the code seems misleading?
2. Is not this code strictly equal to `const targetGridEl = rowEl.closest('.o_grid_mode');` ? :grimacing: `closest` includes the node itself

Reply:
1. I consider that if an inner snippet (non-grid) is in an outer snippet (grid) then it is in a grid in general (not in the sense of "one of the grid children"). But it is true that it is not clear, I will improve the comment :)
2. Indeed... I don't know why I was convinced that `closest` didn't include the node too :facepalm: